pkg/manager/member/pd_scaler.go

```diff
@@ -36,16 +38,29 @@ type pdScaler struct {
 // NewPDScaler returns a Scaler
 func NewPDScaler(pdControl pdapi.PDControlInterface,
 	pvcLister corelisters.PersistentVolumeClaimLister,
-	pvcControl controller.PVCControlInterface) Scaler {
-	return &pdScaler{generalScaler{pdControl, pvcLister, pvcControl}}
+	pvcControl controller.PVCControlInterface,
+	recorder record.EventRecorder) Scaler {
+	return &pdScaler{generalScaler{pdControl, pvcLister, pvcControl, recorder}}
 }
 
 func (psd *pdScaler) Scale(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSet, newSet *apps.StatefulSet) error {
 	scaling, _, _, _ := scaleOne(oldSet, newSet)
-	if scaling > 0 {
-		return psd.ScaleOut(tc, oldSet, newSet)
-	} else if scaling < 0 {
-		return psd.ScaleIn(tc, oldSet, newSet)
+	oldReplicas := *oldSet.Spec.Replicas
+	targetReplicas := *newSet.Spec.Replicas
+	if scaling != 0 {
+		if scaling > 0 {
+			err := psd.ScaleOut(tc, oldSet, newSet)
+			if err != nil {
+				return err
+			}
+		} else if scaling < 0 {
+			err := psd.ScaleIn(tc, oldSet, newSet)
+			if err != nil {
+				return err
+			}
+		}
+		psd.recorder.Event(tc, corev1.EventTypeNormal, scalingEventReason, fmt.Sprintf(scalingEventMsgPattern, "pd", oldReplicas, targetReplicas))
```

Comment:
The updates to the statefulset will only be done [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_member_manager.go#L267), any change before that line is possible to be skipped until next sync as stated in https://github.com/pingcap/tidb-operator/pull/2150#discussion_r407356912.

Reply:
updated, scaling event removed for now.